docs: add two new overview pages: developer-guide-overview and unders… (#44091)
angular/angular: aio/content/guide/understanding-angular-overview.md

# Understanding Angular

To understand the capabilities of the Angular framework, you need to learn about the following:

> * Components

Reviewer: To create a clean format, align the text on the fifth character from the edge \(or indentation\).
Please repeat for all ordered and unordered lists.

```suggestion
*   Components
```

Reviewer 2: I will do a markdown format sweep later. When I add these changes, I get all sorts of squiggly lines and warnings.

> As described in [What Is Angular][AioGuideWhatIsAngular], the Angular framework consists of four basic elements:

Reviewer 3: This statement is a bit confusing to me. Is this a summarization that we lean on elsewhere in the documentation?

For one, this mixes tangible and intangible concepts. Components, Directives, and templates are things that you write, while Dependency Injection is a design pattern that you follow (made easier by Angular's support). Listing Components and Templates separately is also slightly confusing, as you can't have a template without a component or a component without a template - they're two sides of the same concept.

I get that we're trying to say "here are the basic conceptual building blocks of Angular". But the line:

> the Angular framework consists of four basic elements

reads more to me like "Angular is built out of these things".

Reviewer 2: Good point. I rewrote this introduction. Let me know what you think!

> As described in [What Is Angular][AioGuideWhatIsAngular], the Angular frameworks consists of four basic elements:

Reviewer 4: ```suggestion
As described in [What Is Angular][AioGuideWhatIsAngular], the Angular framework consists of four basic elements:
```

* Templates
* Dependency injection
* Directives

> Not every application uses all four elements equally. But understanding these elements can help you build scalable, robust Angular applications.

Reviewer 4: ```suggestion
Not every application uses all four elements equally. But understanding these elements can help you build scalable and robust Angular applications.
```

Reviewer 2: Technically, both versions are correct. :) But I think using `and` in this case makes the content just a touch easier to understand.

## Prerequisites

To get the most out of these developer guides, you should review the following topics:

* [What is Angular][AioGuideWhatIsAngular]
* [Getting started tutorial][AioStart]

## Learn about Angular basics

<div class="card-container">
  <a href="guide/component-overview" class="docs-card" title="Components">
    <section>Components</section>

>     <p>Learn about Angular components; a key building block of Angular development.</p>

Reviewer: ```suggestion
    <p>Learn about Angular components. A component is a key building block of Angular development.</p>
```

    <p class="card-footer">Components</p>
  </a>
  <a href="guide/template-syntax" class="docs-card" title="Templates">
    <section>Forms</section>

>     <p>Learn about building Angular templates.</p>

Reviewer: ```suggestion
    <p>Learn how to build an Angular template.</p>
```

>     <p>Learn about building an Angular template.</p>

Reviewer: ```suggestion
    <p>Learn about how to build an Angular template.</p>
```

Changed to match similar usage.

    <p class="card-footer">Templates</p>
  </a>
  <a href="guide/built-in-directives" class="docs-card" title="Directives">
    <section>Directives</section>

>     <p>Learn about Angular directives: classes that add additional behavior to elements in your Angular applications.</p>

Reviewer: ```suggestion
    <p>Learn about Angular directives. A directive is a class that adds additional behavior to elements in your Angular application.</p>
```

    <p class="card-footer">Directives</p>
  </a>
  <a href="guide/dependency-injection" class="docs-card" title="Dependency injection">
    <section>Dependency injection</section>

>     <p>Learn about dependency injection: services or objects that a class needs to perform its function.</p>

Reviewer: ```suggestion
    <p>Learn about dependency injection. Dependency injection is a service or object that a class needs to perform a function.</p>
```

>     <p>Learn about dependency injection. Dependency injection refers to services or objects that a class needs to perform its function.</p>

Reviewer: ```suggestion
    <p>Learn about dependency injection. Dependency injection refers to services or objects that a class needs to perform a specific function.</p>
```

Removed possessive case to reduce cognitive load.
